plutus-core-spec/plutus-core-specification.tex

```diff
@@ -1371,6 +1378,144 @@ \subsubsection{Comparison of the typed CK machine and the untyped CEK machine}
 closure $C$ with $\discharge{C} = \erase{V}$.
 \end{itemize}
 
+\section{Serialisation}
+
+We use the \texttt{flat} \citep{flat} format to serialise plutus terms. The
+\texttt{flat} format encodes sum types as tagged unions and products by
+concatenating their contents. We proceed by defining the structure and the data
+types of untyped plutus core and how they get serialised.
+
+\subsection{Variable length data}
+
+\texttt{Non-empty lists} are encoded by prefixing the element stored with `0'
+if this is the last element or `1' if there is more data following.
+
+\noindent We encode variable length data as a non-empty list of chunks, 7 bits each,
+with the least significant chunk first and the most significant bit first in the chunk.
+
+\medskip
+\noindent Lets calculate the encoding of the \texttt{32768} index (unsigned, arbitrary
+length integer):
+\begin{enumerate}
+  \item Converting \texttt{32768} to binary: \\
+    \verb|32768| $\rightarrow$ \verb|0b1000000000000000|
+  \item Split into 7 bit chunks: \\
+    \verb|0b1000000000000000| $\rightarrow$ \verb|0000010 0000000 0000000|
+  \item Reorder chunks (least significant chunk first): \\
```

Comment:
Missing the bit ordering step after this?

Reply:
I think I should remove this example, now that I will add the more comprehensive one.

Reply:
There is also, no reordering bits after this. Bits are already properly ordered, this would be the final binary output.